src/Collapse.js

```diff
@@ -1,234 +1,169 @@
+    if (prevState.to !== this.state.to) {
+      this.props.onHeightReady(this.state.to);
     }
 
-    this.setState({height: isOpened || !this.renderStatic ? height : 0});
+    const from = this.wrapper.clientHeight;
+    const to = this.props.isOpened ? this.content.clientHeight : 0;
 
-    const reportHeight = isOpened ? height : 0;
+    if (from !== to) {
+      this.setState({action: RESIZE, from, to});
+      return;
+    }
 
-    if (this.state.height !== reportHeight) {
-      onHeightReady(reportHeight);
+    if (this.state.action === REST) {
+      this.setState({action: STABLE, from, to});
     }
   },
 
 
-  getMotionHeight(height) {
-    const {isOpened, springConfig, fixedHeight} = this.props;
-    const {isOpenedChanged} = this.state;
-
-    const newHeight = isOpened ? Math.max(0, parseFloat(height)).toFixed(1) : stringHeight(0);
-
-    // No need to animate if content is closed and it was closed previously
-    // Also no need to animate if height did not change
-    const skipAnimation = !isOpenedChanged && !isOpened ||
-      this.height === newHeight && fixedHeight === -1;
+  componentWillUnmount() {
+    cancelAnimationFrame(this.raf);
+  },
 
-    const springHeight = spring(isOpened ? Math.max(0, height) : 0, {
-      precision: PRECISION,
-      ...springConfig
-    });
-    const instantHeight = isOpened ? Math.max(0, height) : 0;
 
-    return skipAnimation ? instantHeight : springHeight;
+  onContentRef(content) {
+    this.content = content;
   },
 
 
-  renderFixed() {
-    const {
-      springConfig: _springConfig,
-      onHeightReady: _onHeightReady,
-      onRest: _onRest,
-      isOpened,
-      style,
-      children,
-      fixedHeight,
-      keepCollapsedContent,
-      ...props
-    } = this.props;
-
-    if (this.renderStatic) {
-      this.renderStatic = false;
-      const newStyle = {overflow: 'hidden', height: isOpened ? fixedHeight : 0};
+  onWrapperRef(wrapper) {
+    this.wrapper = wrapper;
+  },
 
-      if (!keepCollapsedContent && !isOpened) {
-        return null;
-      }
-      this.height = stringHeight(fixedHeight);
-      return <div style={{...newStyle, ...style}} {...props}>{children}</div>;
-    }
 
-    return (
-      <Motion
-        defaultStyle={{height: isOpened ? 0 : fixedHeight}}
-        style={{height: this.getMotionHeight(fixedHeight)}}>
-        {({height}) => {
-          this.height = stringHeight(height);
-
-          // TODO: this should be done using onEnd from ReactMotion, which is not yet implemented
-          // See https://github.com/chenglou/react-motion/issues/235
-          if (!keepCollapsedContent && !isOpened && this.height === stringHeight(0)) {
-            return null;
-          }
-
-          const newStyle = {overflow: 'hidden', height};
-
-          return <div style={{...newStyle, ...style}} {...props}>{children}</div>;
-        }}
-      </Motion>
-    );
+  onRest() {
+    this.raf = requestAnimationFrame(() => this.setState({action: REST}));
```

Comment:
Better bind the frame callback once: in the constructor, or use ES7 class property assignment with an arrow function.